daemon/cluster/cluster.go

```diff
@@ -1121,6 +1130,82 @@ func (c *Cluster) RemoveService(input string) error {
 	return nil
 }
 
+// ServiceLogs collects service logs and writes them back to `config.OutStream`
+func (c *Cluster) ServiceLogs(ctx context.Context, input string, config *backend.ContainerLogsConfig, started chan struct{}) error {
+	if !c.isActiveManager() {
```

Comment:
Need to get a readlock for this. You can release it before the long running operations.